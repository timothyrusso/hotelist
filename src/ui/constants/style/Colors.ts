export const Colors = {
	alpha: {
		default: "#C2DFFF",
		hover: "#A9C8F0",
		disabled: "#E3ECF8",
	},
	beta: {
		default: "#FFE2A1",
		hover: "#FFD385",
		disabled: "#F6E8C7",
	},
	gamma: {
		default: "#B7B7FF",
		hover: "#9E9EFF",
		disabled: "#D7D7F5",
	},
	delta: {
		default: "#D8E9AE",
		hover: "#C6DB94",
		disabled: "#E7F1D2",
	},
	black: {
		default: "#1A1A1A",
		hover: "#1A1A1A",
		disabled: "#B8B5B1",
	},
	white: {
		default: "#FFFFFF",
		hover: "#F0F0F0",
		disabled: "#F0F0F0",
	},
} as const;
