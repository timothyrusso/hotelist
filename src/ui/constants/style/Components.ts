export const Components = {
	hotelCard: {
		image: {
			width: 100,
		},
	},
	hotelDetailsPage: {
		image: {
			height: 300,
		},
	},
};
