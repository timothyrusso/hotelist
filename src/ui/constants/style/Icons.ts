export const Icons = {
	star: "star",
	hotel: "hotel",
	location: "location-sharp",
	euro: "euro",
	search: "search",
	options: "options",
	sad: "sad",
	close: "close-a",
	chevronRight: "chevron-right",
	filter: "filter",
} as const;
