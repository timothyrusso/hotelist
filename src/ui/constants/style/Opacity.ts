export const Opacity = {
	default: 0.8,
	disabled: 0.5,
} as const;
