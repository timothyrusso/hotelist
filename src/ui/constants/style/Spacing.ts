export const Spacing = {
	Zero: 0,
	HalfMinimal: 1,
	ThreeQuarterMinimal: 1.5,
	Minimal: 2,
	MinimalDouble: 4,
	Single: 6,
	SingleAndHalf: 8,
	Double: 12,
	Triple: 16,
	TripleAndHalf: 20,
	Fourfold: 24,
	Quintuple: 32,
	Sextuple: 48,
	separator40: 40,
	separator80: 80,
	separator120: 120,
	separator160: 160,
};
