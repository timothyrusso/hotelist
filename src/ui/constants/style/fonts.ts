export const fonts = {
	interRegular: "inter-regular",
	interMedium: "inter-medium",
	interBold: "inter-bold",
} as const;

export const fontsConfig = {
	"inter-regular": require("../../assets/fonts/Inter-Regular.ttf"),
	"inter-medium": require("../../assets/fonts/Inter-Medium.ttf"),
	"inter-bold": require("../../assets/fonts/Inter-Bold.ttf"),
};
