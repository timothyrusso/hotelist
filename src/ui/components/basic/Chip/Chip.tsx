import type { FC } from "react";
import {
	type StyleProp,
	Text,
	type TextStyle,
	View,
	type ViewStyle,
} from "react-native";
import { styles } from "./Chip.style";

type ChipProps = {
	label: string;
	style?: StyleProp<ViewStyle>;
	textStyle?: StyleProp<TextStyle>;
};

export const Chip: FC<ChipProps> = ({ label, style, textStyle }) => {
	return (
		<View style={[styles.container, style]}>
			<Text style={[styles.label, textStyle]}>{label}</Text>
		</View>
	);
};
