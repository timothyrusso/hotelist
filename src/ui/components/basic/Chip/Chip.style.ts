import { Colors } from "@/src/ui/constants/style/Colors";
import { Fonts } from "@/src/ui/constants/style/Fonts";
import { Spacing } from "@/src/ui/constants/style/Spacing";
import { StyleSheet } from "react-native";

export const styles = StyleSheet.create({
	container: {
		padding: Spacing.SingleAndHalf,
		borderRadius: Spacing.Fourfold,
		backgroundColor: Colors.white.default,
	},
	label: {
		fontSize: Spacing.Triple,
		fontFamily: Fonts.interBold,
		color: Colors.black.default,
	},
});
