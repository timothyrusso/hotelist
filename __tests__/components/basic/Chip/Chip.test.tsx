import { Chip } from "@/src/ui/components/basic/Chip/Chip";
import { Colors } from "@/src/ui/constants/style/Colors";
import { Fonts } from "@/src/ui/constants/style/Fonts";
import { Spacing } from "@/src/ui/constants/style/Spacing";
import { render } from "@testing-library/react-native";

describe("Chip", () => {
	beforeEach(() => {
		jest.clearAllMocks();
	});

	describe("snapshot tests", () => {
		it("should render correctly with default props", () => {
			// Render the component with default props
			const { toJSON } = render(<Chip label="Test Chip" />);

			expect(toJSON()).toMatchSnapshot();
		});

		it("should render correctly with custom container style", () => {
			const { toJSON } = render(
				<Chip
					label="Test Chip"
					style={{
						backgroundColor: Colors.alpha.default,
						padding: Spacing.TripleAndHalf,
					}}
				/>,
			);

			// ASSERT: Verify the component renders correctly
			expect(toJSON()).toMatchSnapshot();
		});

		it("should render correctly with custom text style", () => {
			const { toJSON } = render(
				<Chip
					label="Test Chip"
					textStyle={{
						color: Colors.alpha.default,
						fontSize: Spacing.Triple,
						fontFamily: Fonts.interMedium,
					}}
				/>,
			);

			expect(toJSON()).toMatchSnapshot();
		});
	});

	describe("functional tests", () => {
		it("should render the label text correctly", () => {
			const { getByText } = render(<Chip label="Custom Label" />);

			expect(getByText("Custom Label")).toBeTruthy();
		});

		it("should handle multiple style objects in arrays", () => {
			const { toJSON } = render(
				<Chip
					label="Test Chip"
					style={[
						{ backgroundColor: Colors.alpha.default },
						{ padding: Spacing.Double },
					]}
					textStyle={[
						{ color: Colors.white.default },
						{ fontSize: Spacing.Triple },
					]}
				/>,
			);

			// Verify the component renders without errors
			expect(toJSON()).toBeTruthy();
		});
	});

	describe("edge cases", () => {
		it("should handle empty label", () => {
			const { getByText } = render(<Chip label="" />);

			expect(getByText("")).toBeTruthy();
		});
	});
});
