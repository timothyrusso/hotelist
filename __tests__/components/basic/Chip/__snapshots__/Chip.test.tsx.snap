// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip snapshot tests should render correctly with custom container style 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 24,
        "padding": 8,
      },
      {
        "backgroundColor": "#5672bc",
        "padding": 20,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#1A1A1A",
          "fontFamily": "inter-bold",
          "fontSize": 16,
        },
        undefined,
      ]
    }
  >
    Test Chip
  </Text>
</View>
`;

exports[`Chip snapshot tests should render correctly with custom text style 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 24,
        "padding": 8,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#1A1A1A",
          "fontFamily": "inter-bold",
          "fontSize": 16,
        },
        {
          "color": "#5672bc",
          "fontFamily": "inter-medium",
          "fontSize": 16,
        },
      ]
    }
  >
    Test Chip
  </Text>
</View>
`;

exports[`Chip snapshot tests should render correctly with default props 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 24,
        "padding": 8,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#1A1A1A",
          "fontFamily": "inter-bold",
          "fontSize": 16,
        },
        undefined,
      ]
    }
  >
    Test Chip
  </Text>
</View>
`;
